product_packaging_type/models/product_packaging_type.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019 Camptocamp (<http://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from collections import OrderedDict
+from odoo import api, fields, models
+
+
+class ProductPackagingType(models.Model):
+    _name = "product.packaging.type"
+    _description = "Type management for product.packaging"
+    _order = 'sequence, code'
+
+    name = fields.Char(required=True, translate=True)
+    code = fields.Char(required=True)
+    sequence = fields.Integer(required=True)
+    has_gtin = fields.Boolean()
+    # required = fields.Boolean()
+    active = fields.Boolean(default=True)
+    is_default = fields.Boolean()
+
+
+class ProductPackaging(models.Model):
+    _inherit = "product.packaging"
+    _order = 'product_id, sequence'
+
+    @api.model
+    def default_packaging_type_id(self):
+        PackType = self.env['product.packaging.type']
+        types = PackType.search([('is_default', '=', True)], limit=1)
+        if types:
+            return types
+        types = PackType.search([], limit=1)
+        if types:
+            return types
+        return PackType.create(
```

Comment:
Creating something in a default values, is it something advised to do?

Reply:
I didn't want to get into this on my first review, but IMO it's better to have a default record as module data and a constraint to enforce there must be at least one active set as default.